src/ng/compile.js

```diff
@@ -3101,6 +3152,33 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         }
       });
 
+      function recordChanges(key, currentValue, previousValue) {
+        if (isFunction(destination.$onChanges) && currentValue !== previousValue) {
+          // If we have not already scheduled the top level onChangesQueue handler then do so now
+          if (!onChangesQueue) {
+            scope.$$postDigest(flushOnChangesQueue);
```

Comment:
We want all the changes in the current digest to settle before we trigger the onChanges hook.
For this to happen we need the current digest to have completed, which is why postDigest is a good place. Perhaps you can convince me otherwise?